src/views/administration/vuln-sources/VulnSourceOSVAdvisories.vue

```diff
@@ -1,182 +1,47 @@
 <template>
-  <b-card no-body :header="header">
-    <b-card-body>
-      <img alt="OSV logo" src="@/assets/img/osv-logo.png" width="65" />
-      <hr />
-      <c-switch
-        color="primary"
-        id="vulnsourceEnabled"
-        label
-        v-bind="labelIcon"
-        v-model="vulnsourceEnabled"
-        :disabled="enabledEcosystems.length === 0"
-      />
-      {{ $t('admin.vulnsource_osv_advisories_enable') }}
-      <br />
-      <c-switch
-        color="primary"
-        id="aliasSyncEnabled"
-        label
-        v-bind="labelIcon"
-        v-model="aliasSyncEnabled"
-        :title="$t('admin.vulnsource_alias_sync_enable_tooltip')"
-        :disabled="!vulnsourceEnabled"
-      />
-      {{ $t('admin.vulnsource_alias_sync_enable') }}
-      <p class="font-sm text-muted">
-        <span class="fa fa-warning">&nbsp;</span>
-        <a :href="aliasGitHubIssueUrl" target="_blank">{{
-          $t('admin.vulnsource_osv_alias_sync_warning')
-        }}</a>
-      </p>
-      <hr />
-      {{ $t('admin.vulnsource_osv_advisories_desc') }}
-      <hr />
-      <b-validated-input-group-form-input
-        id="nvd-feeds-url"
-        :label="$t('admin.vulnsource_osv_base_url')"
-        input-group-size="mb-3"
-        rules="required"
-        type="text"
-        v-model="osvBaseUrl"
-        lazy="true"
-      />
-      <hr />
-      <b-form-group label="Ecosystems">
-        <div class="list-group" style="width: 40%">
-          <span v-for="ecosystem in enabledEcosystems" :key="ecosystem">
-            <actionable-list-group-item
-              :value="ecosystem"
-              :delete-icon="true"
-              @actionClicked="removeEcosystem(ecosystem)"
-            />
-          </span>
-          <actionable-list-group-item
-            :add-icon="true"
-            @actionClicked="$root.$emit('bv::show::modal', 'ecosystemModal')"
-          />
-        </div>
-      </b-form-group>
-    </b-card-body>
-    <b-card-footer>
-      <b-button
-        :disabled="this.vulnsourceEnabled && !this.osvBaseUrl"
-        variant="outline-primary"
-        class="px-4"
-        @click="saveUrl"
-      >
-        {{ $t('message.update') }}
-      </b-button>
-    </b-card-footer>
-    <ecosystem-modal v-on:selection="updateEcosystem" />
-  </b-card>
+  <div>
+    <extension-config-form
+      extension-name="osv"
+      extension-point-name="vuln.datasource"
+      @show-modal="openModal"
+    />
+    <ecosystem-modal
+      id="osv-ecosystem-modal"
+      :allowed-values="allowedValues"
+      :selected-values="selectedValues"
+      @selection="updateValues"
+    />
```

Comment:
What is the reason for keeping this separate from the string list input? Ideally the modal would come built-in with the input so it's reusable.

The concept of a modal for item selection should be generic enough, it's just that "ecosystems" don't make sense outside of OSV.

Reply:
If possible, yes. It's just that the name `ecosystem-modal` doesn't make sense for anything but OSV.